ugvc/mrd/bqsr_inference_utils.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import annotations
+
+import os
+import subprocess
+from os.path import join as pjoin
+
+import joblib
+import numpy as np
+import pandas as pd
+import pysam
+from joblib import Parallel, delayed
+from tqdm import tqdm
+
+from ugvc.dna.utils import revcomp
+from ugvc.utils.cloud_sync import cloud_sync
+from ugvc.utils.misc_utils import set_pyplot_defaults
+from ugvc.vcfbed.variant_annotation import get_cycle_skip_dataframe, get_motif_around
+
+set_pyplot_defaults()
+
+
+def iterate_vcf_in_chunks(vcf_file, chunk_size, model, output_file, chrstr, params, show_progress=True):
+    print(f"fetching chr {chrstr} from {vcf_file}\n")
+    with pysam.VariantFile(vcf_file) as vcf_in, pysam.VariantFile(output_file, "w", header=vcf_in.header) as vcf_out:
+
```

Comment:
Add this to get the list of contigs (chromosomes) to run inference on so you don't have to hard code them:
contigs = [x.name for x in header.contigs.values() if "_" not in x.name and "-" not in x.name and x.name != "chrM" and x.name != "chrEBV"]

Then pass it on to the inference function.

We could remove the if statements to make it more general, this way the inference will also run on contigs where there are no featuremap entries so you would just have to see it doesn't cause issues.

Reply:
if we remove the ifs on the contigs = ...
need to make sure that the fetch from featuremap on these chrs is empty or process_chunk will not fail